src/test/kotlin/org/rust/ide/highlight/RsRainbowVisitorTest.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.highlight
+
+import org.intellij.lang.annotations.Language
+import org.rust.lang.RsTestBase
+import org.rust.lang.core.psi.RsPatBinding
+import org.rust.lang.core.psi.RsPath
+
+class RsRainbowVisitorTest : RsTestBase() {
+    fun checkRainbow(@Language("Rust") text: String) {
+        InlineFile(text)
+        val path = findElementInEditor<RsPath>("X")
+        val ref = path.reference.resolve() ?: return
+        when (ref) {
+            is RsPatBinding -> myFixture.testRainbow("main.rs", "${ref.referenceName}_${ref.textOffset}", true, true)
+            else -> myFixture.testRainbow("main.rs", path.referenceName, true, false)
+        }
+    }
+
+    fun testPathBinding() = checkRainbow("""
+        fn main() {
+            let mut test = "";
+            test = "";
+            //X
+        }
+    """)
+
+    fun testFunctionBinding() = checkRainbow("""
+        fn foo() {}
+        fn main() {
+            foo();
```

Comment:
Hm, shouldn't we have some `<rainbow>` tags in here? Like this: https://github.com/JetBrains/intellij-community/blob/fb1792bc5d19aa616429a26ac447b7f3cee8f355/java/java-tests/testSrc/com/intellij/java/codeInsight/daemon/LightRainbowHighlightingTest.java#L41

Does this test actually fail if there's a bug in rainbow highlighter? 

I thing we should also add more flawourful tests here, to make sure that we actually handle shadowing right.

Something like

```
fn foo(param: i32) {
   let x = param + param; // the first three params should be colored using the same color;
let y = {
   let param = x + y; // now *first* param should use  use a different color;
};
  param // and this shoulduse the original color. 
}

```